### ISSUE TITLE
Enable/disable SBOM generation for specific language types

### ISSUE THREAD

**Issue by shyam-dev** [anchore/syft#840]
**What would you like to be added**:

The syft command line tool generates SBOM's for all types of languages - JS, Ruby, Python, Php, Golang, Java, Rust & base operating system packages. 

However, it would be nice to provide some command line options to syft, so that we can enable/disable the SBOM generation for specific language types

**Why is this needed**:

We at Deepfence (www.deepfence.io) are big fans of syft & gripe, and we would like to integrate them into our open source offering, ThreatMapper (https://github.com/deepfence/ThreatMapper).

We get requests from community that they know the language types of their images, and would like to run vulnerability scans for those language types only.

If we provide such an option, it would improve the speed of the vulnerability scans.

**Additional context**:

We are glad to provide a pull request for this feature.

**Comment by wagoodman** [comment]
Hey @shyam-dev!

We are definitely interested in allowing for user configurability when it comes to which catalogers to run or not run. We have an existing issue that is fairly close to this (https://github.com/anchore/syft/issues/465) and now a community submitted PR that does something very similar (https://github.com/anchore/syft/pull/843).

Internally we've also chatted about this in the past, but it was really conversation that was in the wrong issue. https://github.com/anchore/syft/issues/516#issuecomment-932299428 . I'd like to chat further here about how this solution could look like and what functionalities we're looking to get out of this.

I'll adjust my proposal from the other issue and re-post it here with the updates soon...

**Comment by shyam-dev** [comment]
Hello @wagoodman - We recently [raised a PR](https://github.com/anchore/syft/pull/843) that solves this issue. 

Happy to talk through the use cases that we saw would be helped by this PR.

**Comment by wagoodman** [comment]
> We recently https://github.com/anchore/syft/pull/843 that solves this issue

Indeed, and the PR looks great given the current state of the catalogers. What I'm wondering about is how we should change the catalogers to better surface a configuration that makes sense to users. 

Today the cataloger names represent what they are searching for (e.g. `rpmdb-cataloger`) but haven't been planned out in terms of collections of things. Before we surface the cataloger names directly as the values for the suggested configuration addition, I think we should look at the names and make certain they make the most sense for this new context. 

I want to put forth some changes based on the previous conversations from #516 that we should consider to the existing catalogers before merging #843 . I think tweaking the existing cataloger names to a more common nomenclature makes sense to do first with the suggested (loose) nomenclature being `<language-or-ecosystem>[-<target-file>]-cataloger`:

- `ruby-gem-spec-cataloger` (specification/**/gem.spec) : a rename of the existing `ruby-gemspec-cataloger`
- `ruby-gem-file-lock-cataloger` (gemfile.lock) : a rename of the existing `ruby-gemfile-cataloger`
- `python-package-cataloger` (wheel/egg) : _no change_
- `python-requirements-cataloger` (requirements.txt) : this is **split** from todays `python-index-cataloger`
- `python-poetry-lock-cataloger` (poetry.lock) : this is **split** from todays `python-index-cataloger`
- `javascript-package-json-cataloger` (package.json) : a rename of the existing `javascript-package-cataloger`
- `javascript-package-lock-cataloger` (package-lock.json) : this is **split** from todays `javascript-lock-cataloger`
- `javascript-yarn-lock-cataloger` (yarn.lock [not nested in node_modules dir]) : this is **split** from todays `javascript-lock-cataloger`
- `os-dpkg-cataloger` : a rename of the existing `dpkgdb-cataloger`
- `os-rpmdb-cataloger` : a rename of the existing `rpmdb-cataloger` 
- `os-apkdb-cataloger` : a rename of the existing  `apkdb-cataloger`
- `java-archive-cataloger` (jar,war,ear) : this is **split** from todays `java-cataloger`
- `jenkins-plugin-cataloger` (jpi,hpi) : this is **split** from todays `java-cataloger`
- `go-mod-file-cataloger` (go.mod): _no change_
- `go-mod-binary-cataloger` (buildinfo section within go binaries): a slight rename from todays `go-module-binary-cataloger`
- `rust-cargo-lock-cataloger` (cargo.lock) : a rename from todays `rust-cataloger`
- `php-composer-installed-cataloger` (installed.json):   _no change_
- `php-composer-lock-cataloger` (composer.lock):   _no change_

This would make the values that users would need to specify more semantically accurate and more consistent across ecosystems. Making this kind of change after we expose the `enabled-cataloger` options would be a breaking change, which we want to try to avoid.

This would allow for partial input of cataloger names, such as just `python` which would enable all python related catalogers. I want to stress that I don't think this kind of "partial matching" work needs to be done now, but this approach would leave this option open to us in the future without making breaking changes, which is nice.

**Comment by wagoodman** [comment]
Additionally (and similar to the comment https://github.com/anchore/syft/pull/843#issuecomment-1054342921 ), how should this feature work for non-package catalogers? The current PR and title of this issue suggests the configurability of package-catalogers only, however, we have more than just package catalogers that could be useful to surface through this configurable (file metadata cataloger, file digest cataloger, file classifier, etc).

**Comment by wagoodman** [comment]
Question: what if someone wants to configure additional catalogers and not replace the defaults? The current PR allows only setting/overriding the set of catalogers used and not to specify additional catalogers.

**Comment by shyam-dev** [comment]
Hello @wagoodman - There are some interesting points here. Permit me a day or two to get back with some responses.

**Comment by wagoodman** [comment]
This has been partially implemented in #843 and being enhanced in #1038 to allow for partial matches:
```bash
syft myimg:latest --catalogers python
# will select all python catalogers, irregardless of the source type
```
